### ISSUE TITLE
Not working anymore after the 4th July update (version 2.10)

### ISSUE THREAD

**Issue by albertominetti** [Mygod/VPNHotspot#170]
In my S7 edge the version 2.9.0 was working perfectly, but after the update to version 2.9.1 it is not working anymore. This is the error show in the GUI: ip rule add  iif swlan0 lookup rmnet0 priority 17900 exited with 2

![Screenshot_20200706-170319_VPN Hotspot](https://user-images.githubusercontent.com/15971462/86608569-04db1180-bfab-11ea-91ae-469b865ea16b.jpg)

**Comment by Mygod** [comment]
Please do.

**Comment by albertominetti** [comment]
Additional details sent by email

**Comment by albertominetti** [comment]
The previous version (2.9.0) works properly

**Comment by Mygod** [comment]
Have you tried downgrading to [v2.9.0](https://github.com/Mygod/VPNHotspot/releases/tag/v2.9.0)? It seems like there is something wrong with your root.

**Comment by Mygod** [comment]
Also what is 4th July update? v2.9.1 was released more than a months ago.

**Comment by Mygod** [comment]
From your log it seems like you are actually running v2.10.4. I see the problem now. Please try [this apk](https://mygod.be/resources/userscloud/llffn8thkbfn).

**Comment by albertominetti** [comment]
I rely on the version numbering and release date on the Play Store, so just few days ago I received the update with version 2.9.1

https://play.google.com/store/apps/details?id=be.mygod.vpnhotspot
![image](https://user-images.githubusercontent.com/15971462/86764691-859c1b00-c048-11ea-971f-f39a0ed6fc86.png)


I will try the new APK in few hours

**Comment by Mygod** [comment]
It is showing v2.9.1 because I have aborted rollout of v2.10.4. If this apk works for you I will release this version to Google play later today.

**Comment by retapic** [comment]
Same as mine on **Redmi 3**:
![Screenshot_20200707-184539](https://user-images.githubusercontent.com/19671942/86807476-95385580-c082-11ea-89c9-ebff0c52564a.png)
![Screenshot_20200707-184653](https://user-images.githubusercontent.com/19671942/86807525-9f5a5400-c082-11ea-96b0-6f9d6e283669.png)
![Screenshot_20200707-184617](https://user-images.githubusercontent.com/19671942/86807607-aed99d00-c082-11ea-8382-03e0920f712e.png)
![Screenshot_20200707-184633](https://user-images.githubusercontent.com/19671942/86807622-b26d2400-c082-11ea-82b8-c93e326cf100.png)

Now I will try version **2.10.4**. I will inform you if all goes well.

**Comment by retapic** [comment]
On version **2.10.5** the same thing:
![Screenshot_20200707-190622](https://user-images.githubusercontent.com/19671942/86812334-9455f280-c087-11ea-834d-3f45be76417f.png)
![Screenshot_20200707-190632](https://user-images.githubusercontent.com/19671942/86812362-9c159700-c087-11ea-8fcf-e1b04e9dbc4a.png)
![Screenshot_20200707-190650](https://user-images.githubusercontent.com/19671942/86812398-a59eff00-c087-11ea-9ab5-6e323c603c45.png)
![Screenshot_20200707-190659](https://user-images.githubusercontent.com/19671942/86812450-b64f7500-c087-11ea-8904-fe56f0ee1a70.png)

**I sent the log from the application to the developer on the email.**

**Comment by DrinkTaiTea** [comment]
I was also having an issue with one device after upgrading to 2.9.1. It is different from yours in that devices could connect but could not access the internet. I tried the nightly version Mygod posted above, which also did not work. However, clearing data and cache appear to have fixed my issue. It is something you may want to try, and then reinstall if that does not solve it.

**Comment by retapic** [comment]
I want to inform you that the situation is the same on versions **1.9.0** and **1.8.9**.
Here is a video proof: https://photos.app.goo.gl/n1XwLpKyzvzi2G2B8 

Just in case, I dropped the log of the application version 1.8.9 on the developer's e-mail.

**Comment by Mygod** [comment]
@DrinkTaiTea seems to be having the same issue as OP so I will be closing this issue and release the new version. If you have a different issue please open a new issue.

P.S. @DrinkTaiTea Was the issue only for Android 9 devices or older?

**Comment by DrinkTaiTea** [comment]
> @DrinkTaiTea seems to be having the same issue as OP so I will be closing this issue and release the new version. If you have a different issue please open a new issue.
> 
> P.S. @DrinkTaiTea Was the issue only for Android 9 devices or older?

My issue was different in that I had no error message. I could connect like before to the Hotspot with VPN in the background, but couldn't get internet access on the connected devices. I have two devices, with the one having problems being Android 8, while the other I believe is also Android 8. The other worked fine on 2.9.1. It is possible if I had cleared cache and data while still on 2.9.1 with the problematic device that it would have worked. However, I did not try that until I updated to 2.10.4 and maintained the same problem. I think it may be a matter of clearing cache and data to rid of old files.
